iroh/src/rpc_protocol.rs

```diff
@@ -1224,6 +1224,7 @@ pub enum ProviderResponse {
     DocGetDownloadPolicy(RpcResult<DocGetDownloadPolicyResponse>),
     DocSetDownloadPolicy(RpcResult<DocSetDownloadPolicyResponse>),
     DocGetSyncPeers(RpcResult<DocGetSyncPeersResponse>),
+    StreamCreated(RpcResult<StreamCreated>),
```

Comment:
You need to have a way to send the message when the stream is created, otherwise there is no way to distinguish between "error on stream creation" and "error producing first item".

Reply:
got it, thanks